src/textual/widgets/_collapsible.py

```diff
@@ -214,3 +217,8 @@ def compose_add_child(self, widget: Widget) -> None:
             widget: A Widget to add.
         """
         self._contents_list.append(widget)
+
+    def _watch_title(self, title: str) -> None:
+        if not self.is_mounted:
+            return
+        self._title.label = title
```

Comment:
I'd argue (I may be wrong) that we should always update `label` in `_title`, it's the collapsible title that should determine whether it wants to render or not.
I can also see that with the current implementation of `CollapsibleTitle`, this isn't exactly easy...

Which triggered some thinking... maybe `CollapsibleTitle` should be a `Static` with two attribute strings that represent the collapsed and non-collapsed labels, and when `label` or `collapsed` are updated the watchers just update those strings and the one that is in use.
This means that the rendering of `CollapsibleTitle` can be cached.

Reply:
Sorry @rodrigogiraoserrao, I'm afraid I don't fully understand what you mean. How would `renderable` be initialised if `CollapsibleTitle` was a subclass of `Static`?

Reply:
Completely untested, but I mean smth like

```
class CollapsibleTitle(Static):
    title = reactive(...)
    collapsed = reactive(...)

    def watch_title(...):
        self.collapsed_title = f"> {title}"
        self.expanded_title = f"v {title}"

    def watch_collapsed(...):
        if collapsed: self.update(self.collapsed_title)
        else: self.update(self.expanded_title)

    ...
```

Reply:
Maybe something for a different PR? Not sure.

Reply:
I've changed `CollapsibleTitle` to `Static`, hopefully I've understood correctly. I'm afraid I still don't fully understand how this works despite tests passing.

Reply:
Maybe you did it already, but reading the source code for `Static` might help.
It's surprisingly short and it'll probably help.

Reply:
Sorry I'm probably just missing something obvious. I understand how `Static` works, but what I don't quite get is how the `renderable` is being initialised here. Aren't watch methods only called if the value changes?

Reply:
@rodrigogiraoserrao I think I need clarification as well.

Reply:
The renderable is initialised when `collapsed = reactive(True)` is set, which triggers its watch method which calls `update`.
I thought this was “obvious”! 🤔

Reply:
No, I don't understand your point at the beginning of this thread. I don't know what you are arguing for. Can you elaborate?

Reply:
It's harder to understand what I was saying if you didn't review the code in its first iteration.
If you think this implementation looks good (with `CollapsibleTitle` inheriting from `Static`), then that means you probably agree with my point.

The `if not self.is_mounted` eventually led me to think `CollapsibleTitle` could inherit from `Static` so that it easily caches its rendering, which is always one of two different strings.

I was also advocating for an attempt at removing that `if` if possible.